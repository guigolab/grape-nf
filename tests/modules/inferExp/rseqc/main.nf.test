nextflow_process {

    name "Test Process inferExp"
    script "modules/inferExp/rseqc/main.nf"
    process "inferExp"

    tag "module"

    setup {
        run("gtfToGenePred") {
            script "modules/inferExp/rseqc/main.nf"
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf.gz")
                """
            }
        }
        run("genePredToBed") {
            script "modules/inferExp/rseqc/main.nf"
            process {
                """
                input[0] = gtfToGenePred.out
                """
            }
        }
    }

    test("Should infer strandedness from BAM (compressed annotation)") {

        when {
            process {
                """
                input[0] = genePredToBed.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should infer strandedness from BAM (uncompressed annotation)") {

        when {
            process {
                """
                input[0] = genePredToBed.out
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
