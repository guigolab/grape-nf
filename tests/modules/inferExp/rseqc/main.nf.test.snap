{
    "Should infer strandedness from BAM (uncompressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "{\"paired\": true, \"exp\": \"MATE2_SENSE\"}"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T18:54:10.824148"
    },
    "Should infer strandedness from BAM (compressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "{\"paired\": true, \"exp\": \"MATE2_SENSE\"}"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T18:54:02.195879"
    }
}
